syntax = "proto3";

package chatbot;

import "google/protobuf/timestamp.proto";

// Base message for chat conversations
message Conversation {
  int32 id = 1;
  string title = 2;
  google.protobuf.Timestamp created_at = 3;
  google.protobuf.Timestamp updated_at = 4;
  int32 user_id = 5;
}

// Message for individual chat messages
message ChatMessage {
  int32 id = 1;
  int32 conversation_id = 2;
  string content = 3;
  bool is_bot = 4;
  google.protobuf.Timestamp timestamp = 5;
  repeated string attachments = 6;
}

// Message for predefined responses
message PredefinedResponse {
  int32 id = 1;
  string trigger_phrase = 2;
  string response_text = 3;
  int32 priority = 4;
}

// Message for training data
message TrainingData {
  int32 id = 1;
  string question = 2;
  string answer = 3;
  string category = 4;
  google.protobuf.Timestamp created_at = 5;
}
